protostar-rust/src/lib.rs

```diff
@@ -317,4 +352,156 @@ version = \"0.1.0\"";
+            },
+            TestConfig {
+                name: "outer::crate2::execute_next_thing".to_string(),
+                available_gas: None,
+            },
+        ];
+
+        let filtered = filter_tests_by_name(Some("do"), mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![TestConfig {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+            },]
+        );
+
+        let filtered = filter_tests_by_name(Some("run"), mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![TestConfig {
+                name: "crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },]
+        );
+
+        let filtered = filter_tests_by_name(Some("thing"), mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![
+                TestConfig {
+                    name: "crate1::do_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "crate2::run_other_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "outer::crate2::execute_next_thing".to_string(),
+                    available_gas: None,
+                },
+            ]
+        );
+
+        let filtered = filter_tests_by_name(Some("nonexistent"), mocked_tests.clone()).unwrap();
+        assert_eq!(filtered, vec![]);
+
+        let filtered = filter_tests_by_name(Some(""), mocked_tests).unwrap();
+        assert_eq!(
+            filtered,
+            vec![
+                TestConfig {
+                    name: "crate1::do_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "crate2::run_other_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "outer::crate2::execute_next_thing".to_string(),
+                    available_gas: None,
+                },
+            ]
+        );
+    }
+
+    #[test]
+    fn filtering_tests_only_uses_name() {
+        let mocked_tests: Vec<TestConfig> = vec![
+            TestConfig {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "outer::crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },
+        ];
+
+        let filtered = filter_tests_by_name(Some("crate"), mocked_tests).unwrap();
+        assert_eq!(filtered, vec![]);
+    }
+
+    #[test]
+    fn filtering_tests_works_without_crate_in_test_name() {
```

Comment:
Does this test check anything? It only shows that there is no error thrown. I would check if it matches against `name: "thing".to_string()`